src/main/java/com/example/bookstore/validation/FieldsValueMatchValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.example.bookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldsValueMatchValidator implements ConstraintValidator<FieldsMatch, Object> {
+    private String field;
+    private String fieldMatch;
+
+    @Override
+    public void initialize(FieldsMatch constraintAnnotation) {
+        this.field = constraintAnnotation.fields()[0];
+        this.fieldMatch = constraintAnnotation.fields()[1];
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        Object fieldValue = new BeanWrapperImpl(value)
+                .getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(value)
+                .getPropertyValue(fieldMatch);
+
+        if (fieldValue != null) {
```

Comment:
you can simplify this 
return fieldValue != null && fieldValue.equals(fieldMatchValue);